oasst-shared/oasst_shared/schemas/inference.py

```diff
@@ -117,6 +117,82 @@ class SamplingParameters(pydantic.BaseModel):
     max_new_tokens: int = 1024
 
 
+class PluginApiType(pydantic.BaseModel):
+    type: str
+    url: str
+    has_user_authentication: bool | None = False
+    # NOTE: Some plugins using this field,
+    # instead of has_user_authentication
+    is_user_authenticated: bool | None = False
+
+
+class PluginAuthType(pydantic.BaseModel):
+    type: str
+
+
+class PluginOpenAPIParameter(pydantic.BaseModel):
+    name: str
+    in_: str
+    description: str
+    required: bool
+    schema_: object
+
+
+class PluginOpenAPIEndpoint(pydantic.BaseModel):
+    path: str
+    type: str
+    summary: str
+    operation_id: str
+    url: str
+    params: List[PluginOpenAPIParameter]
+
+
+class PluginConfig(pydantic.BaseModel):
+    schema_version: str
+    name_for_model: str
+    name_for_human: str
+    description_for_human: str
+    description_for_model: str
+    api: PluginApiType
+    auth: PluginAuthType
+    logo_url: str | None = None
+    contact_email: str | None = None
+    legal_info_url: str | None = None
+    endpoints: List[PluginOpenAPIEndpoint] | None = None
+
+    def __getitem__(self, key: str) -> Any:
+        return getattr(self, key)
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        setattr(self, key, value)
+
+
+class PluginEntry(pydantic.BaseModel):
+    url: str
+    enabled: bool = True
+    plugin_config: PluginConfig | None = None
+    # NOTE: Idea, is to have OA internal plugins as trusted,
+    # and all other plugins as untrusted by default(until proven otherwise)
+    trusted: bool | None = False
+
+
+class PluginExecutionDetails(pydantic.BaseModel):
+    inner_monologue: List[str]
+    final_tool_output: str
+    final_prompt: str
+    final_generation_assisted: bool
+    achieved_depth: int | None = None
+    error_message: str | None = None
+    status: Literal["success", "failure"]
+
+
+class PluginUsed(pydantic.BaseModel):
+    name: str | None = None
```

Comment:
why could name or url be none here? does it make sense ot have a `PluginUsed` with only execution details and nothing else?

Reply:
I think that I had some misconfigured plugins while developing it, and then got the error with name cant be None, just added None, so it can be deleted. But regarding plugins and parsing, there are quite a few weak points, i will address them soon, as i found them while trying various other plugins...

Reply:
I fixed it just half hour ago, will push it later then.